apps/explorer/src/components/OwnedCoins/OwnedCoins.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { useEffect, useState } from 'react';
+
+import OwnedCoinView from './components/OwnedCoinView';
+
+import type { CoinBalance } from '@mysten/sui.js';
+
+import { Heading } from '~/ui/Heading';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Pagination } from '~/ui/Pagination';
+import { Text } from '~/ui/Text';
+
+export const COINS_PER_PAGE: number = 6;
+
+function OwnedCoins({ id }: { id: string }): JSX.Element {
+    const [uniqueCoins, setUniqueCoins] = useState<CoinBalance[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [isFail, setIsFail] = useState(false);
+    const [currentSlice, setCurrentSlice] = useState(1);
+    const rpc = useRpcClient();
+
+    useEffect(() => {
+        setIsFail(false);
+        setIsLoading(true);
+        rpc.getAllBalances({ owner: id })
```

Comment:
This should be in a `useGetAllBalances` hook that uses `useQuery` under the hood